src/main/steps/tasklistresponse/allegations-of-harm-and-violence/SafetyConcernsMapper.ts

```diff
@@ -1,72 +1,177 @@
-            req.session?.userCase?.PRL_c1A_safteyConcerns?.child?.[`${abuse}`].childrenConcernedAbout.join(',');
-        }
+  if (PRL_c1A_concernAboutRespondent) {
+    request.respondent = {};
+    PRL_c1A_concernAboutRespondent.forEach((abuse: string) => {
+      if (PRL_c1A_safteyConcerns?.respondent?.[abuse]) {
+        request.respondent = {
+          ...request.respondent,
+          [abuse]: PRL_c1A_safteyConcerns?.respondent?.[abuse],
+        };
       }
     });
-    (respondent.value.response.safetyConcerns!.c1asafetyconcerns = safetyConcerns),
-      (respondent.value.response.safetyConcerns!.otherconcerns = otherConcerns);
-    respondent.value.response.safetyConcerns!.abductions = Abductions;
-  } else {
-    (respondent.value.response.safetyConcerns!.c1asafetyconcerns = req.session.userCase.PRL_c1A_safteyConcerns ?? {}),
-      (respondent.value.response.safetyConcerns!.otherconcerns = otherConcerns);
-    respondent.value.response.safetyConcerns!.abductions = Abductions;
   }
-  console.log(respondent.value.response.safetyConcerns);
-  return respondent;
+
+  Object.assign(request, {
+    haveSafetyConcerns: PRL_c1A_haveSafetyConcerns,
+    safetyConcernAbout: PRL_c1A_safetyConernAbout,
+    concernAboutChild: PRL_c1A_concernAboutChild,
+    concernAboutRespondent: PRL_c1A_concernAboutRespondent,
+    otherconcerns: {
+      c1AkeepingSafeStatement: PRL_c1A_keepingSafeStatement,
+      c1AsupervisionAgreementDetails: PRL_c1A_supervisionAgreementDetails,
+      c1AagreementOtherWaysDetails: PRL_c1A_agreementOtherWaysDetails,
+      c1AotherConcernsDrugs: PRL_c1A_otherConcernsDrugs,
+      c1AotherConcernsDrugsDetails: PRL_c1A_otherConcernsDrugsDetails,
+      c1AchildSafetyConcerns: PRL_c1A_childSafetyConcerns,
+      c1AchildSafetyConcernsDetails: PRL_c1A_childSafetyConcernsDetails,
+    },
+    abductions: {
+      c1AabductionReasonOutsideUk: PRL_c1A_abductionReasonOutsideUk,
+      c1AchildsCurrentLocation: PRL_c1A_childsCurrentLocation,
+      c1AchildrenMoreThanOnePassport: PRL_c1A_childrenMoreThanOnePassport,
+      c1ApossessionChildrenPassport: PRL_c1A_possessionChildrenPassport,
+      c1AprovideOtherDetails: PRL_c1A_provideOtherDetails,
+      c1ApassportOffice: PRL_c1A_passportOffice,
+      c1AabductionPassportOfficeNotified: PRL_c1A_abductionPassportOfficeNotified,
+      c1ApreviousAbductionsShortDesc: PRL_c1A_previousAbductionsShortDesc,
+      c1ApoliceOrInvestigatorInvolved: PRL_c1A_policeOrInvestigatorInvolved,
+      c1ApoliceOrInvestigatorOtherDetails: PRL_c1A_policeOrInvestigatorOtherDetails,
+      c1AchildAbductedBefore: PRL_c1A_childAbductedBefore,
+    },
+  });
+
+  // data clean up
+  if (PRL_c1A_haveSafetyConcerns === YesOrNo.NO) {
+    request = {
+      haveSafetyConcerns: PRL_c1A_haveSafetyConcerns,
+    };
+  }
+
+  if (
+    !PRL_c1A_safetyConernAbout?.includes(PRL_C1ASafteyConcernsAbout.RESPONDENT) &&
+    !PRL_c1A_concernAboutChild?.includes(PRL_C1AAbuseTypes.WITNESSING_DOMESTIC_ABUSE)
```

Comment:
Same goes here what if no child concerns are present?

Reply:
It is not an issue.